utils/generator.py

```diff
@@ -625,6 +638,9 @@ def interface_to_python_class(
                 "required": bool,
                 "positional": bool,
             ],
+            "buckets_as_dict": "type" # optional, only present in aggregation response
+                                      # classes that have buckets that can have a list
+                                      # or dict representation
```

Comment:
To make typing easier aggregations have `buckets` typed as a list and `buckets_as_dict` typed as a dict. You can use whichever makes sense depending on how yo request your aggregations. It's a compromise solution to avoid having to apply casts on a type that is a union of list and dict buckets.

Reply:
What do you mind with "type" here?

Reply:
"type" here is the type that the `buckets_as_dict` property will be. This is used on the types Jinja template:

```
    @property
    def buckets_as_dict(self) -> Mapping[str, {{ k.buckets_as_dict }}]:
        return self.buckets  # type: ignore
```